src/Stack/Types/Config.hs

```diff
@@ -840,6 +842,8 @@ data ConfigMonoid =
     -- ^ See: 'configExtraIncludeDirs'
     ,configMonoidExtraLibDirs        :: !(Set Text)
     -- ^ See: 'configExtraLibDirs'
+    , configMonoidOverrideGccPath    :: !(First Text)
```

Comment:
I have opened https://github.com/commercialhaskell/stack/issues/2267 with regard to the overly permissive types.

Given that it's a rather tangential point, feel free to change `configMonoidOverrideGccPath` to `First (Path Abs File)` or not.